oggm/workflow.py

```diff
@@ -369,3 +371,78 @@ def inversion_tasks(gdirs):
 
     # Filter
     execute_entity_task(tasks.filter_inversion_output, gdirs)
+
+
+def merge_glacier_tasks(glcdf, main_rgi_ids, filename='climate_monthly',
+                        input_filesuffix=''):
+    """Shortcut function: run all tasks to merge tributaries to a main glacier
+
+    TODO - Automatic search for tributary glaciers
+    TODO - Every tributary should only be used once
+
+    Parameters
+    ----------
+    glcdf: geopandas.GeoDataFrame
+        which contains the main glaciers and the candidates
+    main_rgi_ids: list of str
+        RGI IDs of the main glaciers of interest. Must be in glcdf.
+    filename: str
+        Baseline climate file
+    input_filesuffix: str
+        Filesuffix to the climate file
+
+    Returns
+    -------
+    merged_gdirs: list of merged GlacierDirectories
+
+    """
+    # init and preprocess all glaciers
+    gdirs_all = init_glacier_regions(glcdf)
+    gis_prepro_tasks(gdirs_all)
+
+    # make sure rgi_ids are iteratable
+    main_rgi_ids = utils.tolist(main_rgi_ids)
+
+    # split main glaciers from candidates
+    maindf = glcdf.loc[glcdf.RGIId.isin(main_rgi_ids)].copy()
+    gdirs_main = [gd for gd in gdirs_all if gd.rgi_id in main_rgi_ids]
+
+    # preprocess the main glaciers to the end
+    climate_tasks(gdirs_main)
+    inversion_tasks(gdirs_main)
+    execute_entity_task(tasks.init_present_time_glacier, gdirs_main)
```

Comment:
Intuitively I wouldn't (re)do all the preprocessing tasks here. I thought the workflow would be: 1 prepro some glaciers, 2 merge them, 3 run the merge simulation. This task should start at step 2 and assume the glaciers are preprocessed already.

(note that it's also because of my recent developments which allow to start from a pre-processed state, which will become the default workflow for most users)

Reply:
Yes, my initial idea was to only preprocess the none relevant glacieres as far as neccessary. Purely with performance in mind.

But I agree, this bloats the workflow function. I changed this. For the unlikely case one runs into performance problems because of this, it's easier to write a dedicated workflow.